### DISCUSSION TITLE
How to get  column input value for masking

### DISCUSSION THREAD

**Discussion by rajmetre** [armenak/DataDefender#194] [General]
I am writing an extension, this extension will mask the first letter of the first name and allow masking of a specified column using a hash value. how to take the column input value in the code?

**Comment by vijayaudupa** [comment]
It is possibel with `pass-current-value` attribute.  Here is a code sample from Sample-Project
https://github.com/armenak/DataDefender/blob/master/sample_projects/anonymizer/requirement.xml

```
<plan id="lipsum-similar">
      <function name="Lipsum#similar">
        <argument name="text" type="String" pass-current-value="true"/>
      </function>
    </plan>
```
function `similar` takes one argument `text` for which current value of the column is passed and output is based on the input.

**Comment by armenak** [comment]
Closing the ticket as correct answer was provided by @vijayaudupa. Thank you.
